packages/api/tests/es/test-collections.js

```diff
@@ -0,0 +1,361 @@
+
+  const collections = queryResult.results.map((c) => ({
+    name: c.name,
+    version: c.version,
+    stats: c.stats
+  }));
+  t.deepEqual(collections, [
+    {
+      name: 'coll1',
+      version: '2',
+      stats: {
+        running: 0,
+        completed: 0,
+        failed: 0,
+        total: 0
+      }
+    },
+    {
+      name: 'coll2',
+      version: '1',
+      stats: {
+        running: 0,
+        completed: 0,
+        failed: 0,
+        total: 0
+      }
+    },
+    {
+      name: 'coll1',
+      version: '1',
+      stats: {
+        running: 0,
+        completed: 2,
+        failed: 0,
+        total: 2
+      }
+    },
+    {
+      name: 'coll3',
+      version: '1',
+      stats: {
+        running: 0,
+        completed: 1,
+        failed: 0,
+        total: 1
+      }
+    }
+  ]);
+});
+
+test.serial('query correctly queries collection by date', async (t) => {
+  const collectionSearch = new Collection({
+    queryStringParameters: {
+      updatedAt__from: (new Date(2020, 0, 25)).getTime(),
+      updatedAt__to: (new Date(2020, 0, 30)).getTime()
+    }
+  }, null, process.env.ES_INDEX);
+  const queryResult = await collectionSearch.query();
+
+  t.is(queryResult.meta.count, 1);
+  t.is(queryResult.results[0].name, 'coll3');
+});
+
+test.serial('aggregateActiveGranuleCollections returns only collections with granules', async (t) => {
+  const collectionSearch = new Collection({}, null, process.env.ES_INDEX);
+  const queryResult = await collectionSearch.aggregateActiveGranuleCollections();
+
+  t.deepEqual(queryResult, ['coll1___1', 'coll3___1']);
+});
+
+test.serial('aggregateActiveGranuleCollections respects date range for granules', async (t) => {
+  const collectionSearch = new Collection({
+    queryStringParameters: {
+      updatedAt__from: (new Date(2020, 1, 25)).getTime(),
+      updatedAt__to: (new Date(2020, 1, 30)).getTime()
+    }
+  }, null, process.env.ES_INDEX);
+  const queryResult = await collectionSearch.aggregateActiveGranuleCollections();
+
+  t.deepEqual(queryResult, ['coll3___1']);
+});
+
+test.serial('queryCollectionsWithActiveGranules returns collection info and stats', async (t) => {
+  const collectionSearch = new Collection({}, null, process.env.ES_INDEX);
+  const queryResult = await collectionSearch.queryCollectionsWithActiveGranules();
+
+  t.is(queryResult.meta.count, 2);
+
+  const collections = queryResult.results.map((c) => ({
+    name: c.name,
+    version: c.version,
+    stats: c.stats
+  }));
+  t.deepEqual(collections, [
+    {
+      name: 'coll1',
+      version: '1',
+      stats: {
+        running: 0,
+        completed: 2,
+        failed: 0,
+        total: 2
+      }
+    },
+    {
+      name: 'coll3',
+      version: '1',
+      stats: {
+        running: 0,
+        completed: 1,
+        failed: 0,
+        total: 1
+      }
+    }
+  ]);
+});
+
+test.serial('queryCollectionsWithActiveGranules respects granule update times, but not collection', async (t) => {
```

Comment:
The test proves what it says but I couldn't figure out why granule update times are respected and not collection

Reply:
Ah, because `queryCollectionsWithActiveGranules` is actually searching for a type of `granule`, not `collection`